.github/workflows/check.yml

```diff
@@ -1,13 +1,20 @@
 name: Check
-on: push
+on: [ pull_request, push ]
 jobs:
-  build:
+  check:
     runs-on: ubuntu-latest
+    # push: always run.
+    # pull_request: run only when the PR is submitted from a forked repository, not within this repository.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    strategy:
+      fail-fast: false
     steps:
-    - uses: actions/checkout@v1
-    - name: Set up JDK 1.8
-      uses: actions/setup-java@v1
+    - uses: actions/checkout@v3
+    - name: Set up OpenJDK 8
```

Comment:
Minor: 
Does `temurin` JDK the kind of OpenJDK?

Reply:
Yes. The latest Oracle JDK "8" is not provided there, then we needed another distribution. `temurin` is close to the standard one, instead of `adopt`. See: https://github.com/actions/setup-java#supported-distributions